test/demo/contextChannels/join.inttest.ts

```diff
@@ -563,3 +564,24 @@ describe('When using a non-directory app', () => {
         }]);
     });
 });
+
+describe('When the provider is reloaded', () => {
+    setupStartNonDirectoryAppBookends(testAppNotInDirectory1);
+
+    test('The client rejoins the channel it was previously in', async () => {
+        const greenChannel = await fdc3Remote.getChannelById(testAppNotInDirectory1, 'green');
+        await greenChannel.join();
+
+        await reloadProvider();
+        await expect(fdc3Remote.getCurrentChannel(testAppNotInDirectory1)).resolves.toHaveProperty('channel', greenChannel.channel);
+        await expect(greenChannel.getMembers()).resolves.toEqual([{uuid: testAppNotInDirectory1.uuid, name: testAppNotInDirectory1.name}]);
+        await reloadProvider();
+        await expect(fdc3Remote.getCurrentChannel(testAppNotInDirectory1)).resolves.toHaveProperty('channel', greenChannel.channel);
+        await expect(greenChannel.getMembers()).resolves.toEqual([{uuid: testAppNotInDirectory1.uuid, name: testAppNotInDirectory1.name}]);
+        await reloadProvider();
+        await expect(fdc3Remote.getCurrentChannel(testAppNotInDirectory1)).resolves.toHaveProperty('channel', greenChannel.channel);
+        await expect(greenChannel.getMembers()).resolves.toEqual([{uuid: testAppNotInDirectory1.uuid, name: testAppNotInDirectory1.name}]);
```

Comment:
If this is just doing the same thing 3 times, would put it in a for loop to make that obvious